torch/distributed/rpc/api.py

```diff
@@ -260,6 +261,14 @@ def shutdown(graceful=True):
         # resolved.
         _cleanup_python_rpc_handler()
         _reset_current_rpc_agent()
+    try:
+        # This API removes files. There is chance of failure doing IO.
+        instantiator.cleanup_generated_modules()
+    except Exception:
+        import traceback
+
+        print("Encountered exception in instantiator.cleanup_generated_modules()")
+        traceback.print_exc()
```

Comment:
Can we do this using `atexit` instead: https://docs.python.org/3.6/library/atexit.html? This is more reliable since the user might forget to call shutdown() or shutdown() might not be called due to a bug in the user's code.

Reply:
Good idea! Didn't know this mechanism before. Updating.